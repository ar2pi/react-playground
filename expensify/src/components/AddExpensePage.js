import React from 'react';

const AddExpensePage = () => <div>Valid component 2</div>;

export default AddExpensePage;