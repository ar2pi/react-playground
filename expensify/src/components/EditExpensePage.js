import React from 'react';

const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>Valid component 3, id: {props.match.params.id}</div>
    );
};

export default EditExpensePage;