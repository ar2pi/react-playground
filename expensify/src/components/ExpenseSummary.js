import React, { Component } from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';

export class ExpenseSummary extends Component {
    nExpenses = () => {
        let s = this.props.expenses.length > 1 ? 's' : '';
        return `${this.props.expenses.length} expense${s}`;
    };
    totExpenses = () => {
        return numeral(getExpensesTotal(this.props.expenses) / 100).format('$0,0.00');
    };
    render() {
        return (
            <div>
                Viewing {this.nExpenses()} totalling {this.totExpenses()}
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    expenses: state.expenses
});

export default connect(mapStateToProps)(ExpenseSummary);
