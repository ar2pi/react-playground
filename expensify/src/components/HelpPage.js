import React from 'react';

const HelpPage = () => <div>Valid component 4</div>;

export default HelpPage;