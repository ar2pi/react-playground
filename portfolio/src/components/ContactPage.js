import React from 'react';

const Contact = () => (
    <div>
        Send an email: <a href="mailto:arthurpicerna@gmail.com">arthurpicerna@gmail.com</a>
    </div>
);

export default Contact;