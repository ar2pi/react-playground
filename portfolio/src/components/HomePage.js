import React from 'react';

const HomePage = () => (
    <div>
        <p>Welcome home</p>
    </div>
);

export default HomePage;