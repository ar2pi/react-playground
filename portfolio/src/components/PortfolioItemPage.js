import React from 'react';

const PortfolioItemPage = (props) => (
    <div>
        <p>Awesome project nÂ°{props.match.params.id}</p>
    </div>
);

export default PortfolioItemPage;