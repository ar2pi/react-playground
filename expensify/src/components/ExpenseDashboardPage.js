import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseSummary from './ExpenseSummary';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseSummary />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;
