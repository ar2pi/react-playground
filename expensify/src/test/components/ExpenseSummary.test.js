import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<ExpenseSummary expenses={expenses} />);
});

test('should summarize multiple expenses', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toBe('Viewing 3 expenses totalling $1,141.95');
});

test('should summarize single expense', () => {
    wrapper.setProps({
        expenses: [expenses[0]]
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toBe('Viewing 1 expense totalling $1.95');
});
