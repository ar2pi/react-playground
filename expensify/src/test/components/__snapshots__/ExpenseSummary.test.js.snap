// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should summarize multiple expenses 1`] = `
<div>
  Viewing 
  3 expenses
   totalling 
  $1,141.95
</div>
`;

exports[`should summarize single expense 1`] = `
<div>
  Viewing 
  1 expense
   totalling 
  $1.95
</div>
`;
